src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -331,7 +324,14 @@ private static ILogger CreateLogger(bool verbose)
                     });
             });
 
+            if (reportFormat == ReportFormat.Sarif && reportWriter != null)
```

Comment:
I personally think we shouldn't check `reportWriter` for null here - since it's a private method, it can only be called from within this class, so we should just make sure it always has a value when calling the method (which it does).